server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -1532,6 +1534,150 @@ public void testLimitNumberOfAffectedResources() {
+            List.of()
+        );
+        var service = createShardsAvailabilityIndicatorService(clusterState);
+
+        HealthIndicatorResult result = service.calculate(true, HealthInfo.EMPTY_HEALTH_INFO);
+        assertThat(
+            result,
+            equalTo(
+                createExpectedResult(
+                    GREEN,
+                    "This cluster has 1 unavailable primary shard.",
+                    Map.of("unassigned_primaries", 1, "started_primaries", 1),
+                    List.of(),
+                    List.of(
+                        new Diagnosis(ACTION_CHECK_ALLOCATION_EXPLAIN_API, List.of(new Diagnosis.Resource(INDEX, List.of(restoredIndex))))
+                    )
+                )
+            )
+        );
+    }
+
+    public void testShouldBeRedWhenFrozenIndexIsUnassignedAndOriginalIsUnavailable() {
+        String originalIndex = "logs-2023.07.11-000024";
+        String restoredIndex = "restored-logs-2023.07.11-000024";
+        List<IndexMetadata> indexMetadata = new ArrayList<>(2);
+        List<IndexRoutingTable> routes = new ArrayList<>(2);
+        indexMetadata.add(
+            IndexMetadata.builder(restoredIndex)
+                .settings(
+                    Settings.builder()
+                        .put(IndexMetadata.SETTING_VERSION_CREATED, IndexVersion.current())
+                        .put(SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_INDEX_NAME_SETTING_KEY, originalIndex)
+                        .put(IndexModule.INDEX_STORE_TYPE_SETTING.getKey(), SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_STORE_TYPE)
+                        .put(SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_PARTIAL_SETTING_KEY, randomBoolean())
+                        .build()
+                )
+                .numberOfShards(1)
+                .numberOfReplicas(0)
+                .build()
+        );
+        routes.add(index(restoredIndex, new ShardAllocation(randomNodeId(), UNAVAILABLE)));
+        // When original does not exist
+        {
+            var clusterState = createClusterStateWith(indexMetadata, routes, List.of(), List.of());
+            var service = createShardsAvailabilityIndicatorService(clusterState);
+
+            HealthIndicatorResult result = service.calculate(true, HealthInfo.EMPTY_HEALTH_INFO);
+            assertThat(
+                result,
+                equalTo(
+                    createExpectedResult(
+                        RED,
+                        "This cluster has 1 unavailable primary shard.",
+                        Map.of("unassigned_primaries", 1),
+                        List.of(
+                            new HealthIndicatorImpact(
+                                NAME,
+                                ShardsAvailabilityHealthIndicatorService.PRIMARY_UNASSIGNED_IMPACT_ID,
+                                1,
+                                "Cannot add data to 1 index [" + restoredIndex + "]. Searches might return incomplete results.",
```

Comment:
Should we review the message in this context, given that the mounted index is consistently in a read-only state. Perhaps we could focus on mentioning how the search is impacted instead?
Similarly, the impacts should not contain INGEST ?

Reply:
About the impacts I agree, actually, I took it a bit further, I did not add any impacts here because the user is not experiencing any search or ingest impact.